sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -284,15 +284,15 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
       url: String,
       table: String,
       columnName: String,
-      lowerBound: Long,
-      upperBound: Long,
+      lowerBound: String,
+      upperBound: String,
```

Comment:
BTW, this suggestion is a breaking change. We need to accept both one if we want to proceed.

cc @cloud-fan and @gatorsmile